### DISCUSSION TITLE
How are `~[dirs]` implemented?

### DISCUSSION THREAD

**Discussion by psprint** [marlonrichert/zsh-snap#230] [Q&A] [answered]
When I try `hash -d '[dir]=/…'` I get:
```
hash: invalid character in directory name: [dir]
```
I wonder how you've managed to make it work?

**Comment by marlonrichert** [comment, marked as answer]
You can read the code [here](https://github.com/marlonrichert/zsh-snap/blob/main/functions/..znap.dirname). Let me know if you have any questions; I can add more comments. 🙂
